domain/src/main/java/org/oppia/android/domain/oppialogger/OppiaLogger.kt

```diff
@@ -10,13 +12,59 @@ import javax.inject.Inject
 class OppiaLogger @Inject constructor(
   private val analyticsController: AnalyticsController,
   private val consoleLogger: ConsoleLogger,
-  private val oppiaClock: OppiaClock
+  private val oppiaClock: OppiaClock,
+  private val appHealthMetricController: AppHealthMetricController
 ) {
   /** Logs high-priority events. See [AnalyticsController.logImportantEvent] for more context. */
   fun logImportantEvent(eventContext: EventLog.Context) {
     analyticsController.logImportantEvent(oppiaClock.getCurrentTimeMs(), eventContext)
   }
 
+  /**
+   * Logs high-priority performance metrics. See [AppHealthMetricController.logHighPriorityMetricEvent]
+   * for more context.
+   */
+  fun logHighPriorityPerformanceMetric(
```

Comment:
I think I mentioned this in an earlier PR, but I think we should be structuring loggers less like OppiaLogger and more like https://github.com/oppia/oppia-android/blob/develop/domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/LearnerAnalyticsLogger.kt where common logs are put together in a single logger moving forward. It seems more maintainable, easier to integrate (since we can tie things to specific scopes a bit more clearly), and possibly easier to test.

Reply:
Yep, modified. Since learner analytics was based on analytics controller, I've introduced a similar PerformanceMetricsController and then used the PerformanceMetricsLogger to log metrics using that controller.